data-source/json/test_relationships.json

```diff
@@ -167,31 +167,351 @@
                             "question": {
                                 "id": "relationship-question",
                                 "type": "General",
-                                "title": "Thinking of {first_person_name}, {second_person_name} is their <em>...</em>",
+                                "title": {
+                                    "text": "Thinking of {first_person_name}, {second_person_name} is their <em>...</em>",
+                                    "placeholders": [
+                                        {
+                                            "placeholder": "first_person_name",
+                                            "transforms": [
+                                                {
+                                                    "transform": "concatenate_list",
+                                                    "arguments": {
+                                                        "list_to_concatenate": {
+                                                            "source": "answers",
+                                                            "identifier": ["first-name", "last-name"],
+                                                            "list_item_selector": "list_item_id"
```

Comment:
Are we decided on 'list_item_selector'? We use 'id_selector' or 'id' within an object within when rules to indicate which id to select.

Reply:
We could use the same format as the when rules:
```
{
	"list_item_selector": { 
		"source": "location",
		"id": "list_item_id"
	}
}

Reply:
This was my thinking - currently it isn't immediately from the arguments the source of the id.

Reply:
I'll make the change in the validator pr as well